src/ext/php7/engine_hooks.c

```diff
@@ -73,17 +75,35 @@ static bool _dd_should_trace_call(zend_execute_data *execute_data, zend_function
 }
 
 static void _dd_copy_function_args(zend_execute_data *call, zval *user_args) {
-    uint32_t i;
+    uint32_t i, first_extra_arg;
     zval *p, *q;
     uint32_t arg_count = ZEND_CALL_NUM_ARGS(call);
 
     // @see https://github.com/php/php-src/blob/PHP-7.0/Zend/zend_builtin_functions.c#L506-L562
```

Comment:
I wonder why the builtin we looked at doesn't handle variadics.

Reply:
The builtin does handle variadics, but we removed it for our case. We didn't need it because we used to forward the call with `zend_call_function()` which apparently doesn't store the variadic args in the same way as an unaltered call. This is just restoring the args copying back to the original `func_get_args()` behavior.